backend/dataall/modules/redshift_datasets_shares/db/redshift_share_object_repositories.py

```diff
@@ -34,7 +34,7 @@ def count_redshift_principal_shares(session, environment_uri: str, group_uri: st
         share_item_shared_states = ShareStatusRepository.get_share_item_shared_states()
         return (
             session.query(ShareObject)
-            .join(
+            .outerjoin(
```

Comment:
So are we using `outerjoin()` here not to include Shares with no Share Object Items? Would that work considering we are filtering by Share object Items type of `RedshiftTable` and for status in one of the shared states?

Also, looking at where we execute `count_redshift_principal_shares` --> i.e. where we call `EnvironmentResourceManager.count_group_resources(...)` it is for both `delete_env` and `remove_group` on env

While it is okay if we do not count share objects with no shared items for `deleve_env` because we execute `EnvironmentResourceManager.delete_env(...)` and clean up shares in other states afterwards - we do not do the same for `remove_group`

Meaning I believe there would be orphaned share object resources when removing an env group in that case